### PR TITLE
Intgeration with official Mongo driver 2.0.0.

### PR DESCRIPTION
**Pull request by durran**
Will remove the WIP tag and rebase against master once the work is ready for review.

### CODE REVIEW

**Comment by muichkine** [comment]
:+1:

**Comment by arthurnn** [comment]
@durran :heart: .
Would this be 5.0 , or 4.1 ?

**Comment by durran** [comment]
I'm calling it 5.0 since there are going to be some API changes.

**Comment by rpechayr** [comment]
+1

**Comment by CarlosPlusPlus** [comment]
+1

**Comment by rpechayr** [comment]
+2

-- 
Romain Péchayre
Co-founder and CTO Applidget
+33681544071
@rpechayr

2015-03-04 21:42 GMT+01:00 Carlos Lazo <notifications@github.com>:

**Comment by fedenusy** [comment]
+1 this is great!

**Comment by Fudoshiki** [comment]
+1

**Comment by rpechayr** [comment]
Any progress on this ? How can we help ?

**Comment by estolfo** [comment]
Yes, we are working on it. Mongoid with the new Ruby driver (mongo gem 2.0 series) will be available soon.

**Comment by rpechayr** [comment]
Great  thank you! It looks like Mongoid is silently becoming an official MongoDB driver :clap: :clap:. Ready to add a 3rd :clap: if you guys give us an approximate ETA.

**Comment by estolfo** [comment]
hard to say exactly but *very soon*

**Comment by rpechayr** [comment]
great

**Comment by stefanhorning** [comment]
:+1:

**Comment by simonmorley** [comment]
Is this anywhere near production ready? We'd be happy to push to a few machines as we're struggling with moped and the connection pool issue.

**Comment by diziet** [comment]
+1 on what @simonmorley says (if you mean the 20100 limit)

**Comment by simonmorley** [comment]
@diziet My errors are more related with moped (https://github.com/mongoid/moped/issues/345)

Can't use the slaves to read right now - seemly struggling for a while but had some how missed.

**Comment by Fudoshiki** [comment]
please tell about planning release date

**Comment by blasterpal** [comment]
:beers: (actually whiskey) for @durran

**Comment by Fudoshiki** [comment]
Can we help anything?

**Comment by rpechayr** [comment]
Yes, maybe sharing what are the next step could help people be productive for this highly important milestone. @estolfo @durran @arthurnn any ideas ?

**Comment by fedenusy** [comment]
Any chance we can get a *very rough* ETA (days, weeks, or months)? Trying to figure out whether it's worth spending time patching moped.

**Comment by matsimitsu** [comment]
I'd love to know an ETA as well, patching Moped / fighting it's quirks is taking quite a chunk of our development time.

**Comment by Fudoshiki** [comment]
Dear, @estolfo @durran @arthurnn, please tell that this branch alive

**Comment by durran** [comment]
This branch is still in progress - I am just switching between this and triaging issues for 5.0.0 at the moment... These two things are top priority as once this branch comes in and we get a 5.0.0 out the door Mongoid will become an officially supported MongoDB product from 5.0.0 on.

**Comment by Fudoshiki** [comment]
Many thanks for the answer

**Comment by rpechayr** [comment]
Great news @durran, the MongoDB inc support is what I expected form seeing that you were back in mongoid after being hired by them. I can't wait to go away from Moped. Today again I had a serious problem were several rails apps (actually 4 different rails apps) could ne properly handle a MongoDB Replica Set failover. They were trying to connect to a dead host for 10 seconds, whereas other hosts were alive.

**Comment by rpechayr** [comment]
To illustrate my last message here is the kind of problems I have with moped : 
https://github.com/mongoid/moped/issues/373.
Before that, we posted a [question](https://github.com/mongoid/moped/issues/332) related to 2-phase commit on Moped and never got any answer.

**Comment by drobune** [comment]
:+1:

**Comment by jarthod** [comment]
\o/

**Comment by rpechayr** [comment]
:clap: x 2^2048

**Comment by drobune** [comment]
:beer:

**Comment by djdarkbeat** [comment]
Amen

**Comment by thijsc** [comment]
Yay! :+1:

**Comment by mrpoundsign** [comment]
:+1: Thanks so much!

**Comment by blasterpal** [comment]
@durran Thanks for all the work. Looking forward to implementing.

**Comment by dedene** [comment]
:+1: Awesome, thanks a lot!!

**Comment by fedenusy** [comment]
Happy Dance

**Comment by orendon** [comment]
good job, thanks :+1:

File: lib/mongoid/contextual/geo_near.rb [hunk 1]
```diff
@@ -246,7 +246,7 @@ def documents
       #
       # @since 3.0.0
       def results
```
Comment (sescobb27) [inline lib/mongoid/contextual/geo_near.rb]: should be called `result` instead of `results` just because is returning one element, isn't it?

**Comment by matuag** [comment]
:+1:

**Comment by deepredsky** [comment]
thanks :+1:

File: lib/mongoid/contextual/map_reduce.rb [hunk 2]
```diff
@@ -299,7 +299,7 @@ def output_collection
       # @since 3.0.0
       def results
```
Comment (sescobb27) [inline lib/mongoid/contextual/map_reduce.rb]: should be called result instead of results just because is returning one element, the same as in geo_near.rb

**Comment by bmurzeau** [comment]
congrats :+1:

**Comment by mdouchement** [comment]
:+1:

**Pull request merged**